server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1228,7 +1239,6 @@ private boolean doMaintain(final long hostId) {
 
         ActionEventUtils.onStartedActionEvent(CallContext.current().getCallingUserId(), CallContext.current().getCallingAccountId(), EventTypes.EVENT_MAINTENANCE_PREPARE, "starting maintenance for host " + hostId, true, 0);
         _agentMgr.pullAgentToMaintenance(hostId);
-        setHostMaintenanceRetries(host);
```

Comment:
Should this be removed? Would it cause any regression?

Reply:
We have removed the global config and so the method is not needed anymore.